### ISSUE TITLE
[ISSUE] Issue with `databricks_schemas` resource when owner is changed manually

### ISSUE THREAD

**Issue by VOVELEE** [databricks/terraform-provider-databricks#2844]
### Configuration
```hcl
resource "databricks_schema" "schema" {
  for_each = var.schemas

  catalog_name = "var.catalog_name"
  name         = each.key
  owner        = local.admin_group
  storage_root = each.value.location != null ? "abfss://${each.value.location.container}@st${var.stage}${each.value.location.storage}.dfs.core.windows.net/${each.value.location.path}" : null

  properties = {
    owner = "root"
  }
}
```

### Expected Behavior
- When somebody manually changes owner of the catalog -> the provider to be able to change back the owner if the user is an admin of the workspace
- This was running smoothly in the past (last tested with 1.18)

### Actual Behavior
- In provider version 1.29 I get the following error (names of catalog and schemas are changed)
```
Error: cannot update schema: User is not an owner of Schema 'catalogA.schema1'. 
```
- Reason behind is the payload the provider sends in the PATCH request. In 1.29 (and may be lower version) it provides the name of the schema although it is not changed. 

### Terraform and provider versions
- Works in 1.18
- Does not work in 1.29

### Debug Output
1.18 (works)

```
2023-10-26T14:43:54.333+0300 [DEBUG] provider.terraform-provider-databricks_v1.18.0.exe: PATCH /api/2.1/unity-catalog/schemas/catalogA.schema1
> {
>   "owner": "group_name_1"
> }
< HTTP/2.0 200 OK
```

1.29 (does NOT work)

```
2023-10-26T14:17:11.398+0300 [DEBUG] provider.terraform-provider-databricks_v1.29.0.exe: PATCH /api/2.1/unity-catalog/schemas/catalogA.schema1
> {
>   "name": "schema1",
>   "owner": "group_name_1",
>   "properties": {
>     "owner": "root"
>   }
> }
< HTTP/2.0 403 Forbidden
```

**Comment by VOVELEE** [comment]
anyone?

**Comment by nkvuong** [comment]
@VOVELEE this is a similar issue to #2778 - there is some work to fix the underlying issue in the OpenAPI spec/Go SDK

**Comment by VOVELEE** [comment]
Fixed in 1.33.0 and upper
